public/js/clients/chrome.js

```diff
@@ -28,6 +42,58 @@ function chromeTabs(callback) {
   });
 }
 
+function onConnection(connection) {
+  const ws = connection._socket;
+  connectionAgents = connection._agents;
+  ws.onopen =function() {
+    messages.forEach((msg) => {
+      console.log("request: ", msg);
+      ws.send(msg)
+    })
+  };
+
+  ws.onmessage =function(e) {
+    console.log("response: ", e.data)
+  };
+
+  const debuggerAgent = getAgent("Debugger")
+  debuggerAgent.enable();
+  debuggerAgent.setPauseOnExceptions({state: "none"});
+  debuggerAgent.setAsyncCallStackDepth({maxDepth: 0});
+
+  const runtimeAgent = getAgent("Runtime");
+  runtimeAgent.enable();
+  runtimeAgent.run();
+}
+
+function debugTab(tab) {
+  function toTitleCase() {
+    return this.substring(0, 1).toUpperCase() + this.substring(1);
+  }
+
+  // NOTE: this is a serious performance issue.
+  // If we decide to productionize the chrome debugger,
+  // we will want to build the bootstrap commands in advance require them in.
+
+  // NOTE: toTitleCase is temporarily monkey patched onto String.prototype
+  // to conform to what generateCommands expects. I
+  // If String.prototype.toTitleCase is defined somwhere else,
+  // it will be replaced.
+  String.prototype.toTitleCase = toTitleCase;
+  let code = generateCommands(protocol);
+  delete String.prototype.toTitleCase;
+  eval(code);
```

Comment:
Can you file an issue to figure out a way to avoid both `delete` and `eval`? It does seem like something we should do sooner rather than later. 

I wonder if shadowing `String` would work?

```
const _String = String;
function String() { return _String.apply(this, arguments); }
String.prototype = Object.create(_String);
String.prototype.toTitleCase = ...
```

Not sure if that's totally correct or would even work...